presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -198,11 +198,9 @@ private void pageOutput(OutputFormat format, List<String> fieldNames)
             if (!pager.isNullPager()) {
                 // ignore the user pressing ctrl-C while in the pager
                 ignoreUserInterrupt.set(true);
-                Thread clientThread = Thread.currentThread();
                 pager.getFinishFuture().thenRun(() -> {
                     userAbortedQuery.set(true);
                     ignoreUserInterrupt.set(false);
-                    clientThread.interrupt();
```

Comment:
I prefer the first approach since it is simpler and less risky (this code has no tests).